### ISSUE TITLE
Using pairs to create columns with existing variable names

### ISSUE THREAD

**Issue by pdeffebach** [JuliaData/DataFramesMeta.jl#117]
I've posted a few issues about the limitations of having keyword arguments used to create columns. Namely #87. 

In https://github.com/JuliaData/DataFrames.jl/pull/1601#issuecomment-440044138, I recently learned that you can use pairs as keyword arguments for most functions. 

```
function foo(;kwargs...)
       return kwargs
end

x = :a 

foo(; x => 1)
```

This seems like an obvious solution to my problem with scoping. It doesn't currently work in DataFramesMeta, but it should not be too hard to add this functionality. Something like 

```
function clean(df, x::Symbol, y::Symbol)
    @transform(df; x => y + 1)
end
```

Obviously figuring our escaping rules on the rhs would be difficult. But its worth keeping this in mind for future work.

**Comment by pdeffebach** [comment]
Update: 

This actually quite hard, see this [pdicourse issue](https://discourse.julialang.org/t/weird-macro-behavior-trying-to-simulate-keyword-arguments/17800/9) I made on the subject. Julia puts function keyword arguments at the front of the argument list, so we have to write a lot of conditionals to deal with it. 

I think any PR to work with function calls in this way might be beyond my abilities, but will give it a shot, if only so that others can jump in.

**Comment by pdeffebach** [comment]
Closed via #171
